website/blog/2024-02-29-tips-useful-features.md

```diff
@@ -0,0 +1,225 @@
+	{
+		$mimes['svg']  = 'image/svg+xml';
+		$mimes['zip']  = 'application/zip';
+		$mimes['json'] = 'application/json';
+
+		return $mimes;
+	}
+```
+
+And register the filter:
+
+```php
+\add_filter('upload_mimes', [$this, 'enableMimeTypes']);
+```
+
+This class also supports converting your media into **_WebP_** format, but more on that below.
+
+## WebP conversion
+
+The WebP file format is becoming more and more popular, with its smaller file sizes and better compression, it is a preferred replacement to .jpg and .png formats. Eightshift DevKit supports converting your existing files to WebP format.
+
+If you already have the Media class in your project, you need to run this WP-CLI command:
+
+```bash
+wp boilerplate create webp-media-column
+```
+
+This command will add a new column in your ***List view*** in Media Library that shows if the media is converted to WebP or not.
+
+![WebP column in Media Library](/img/blog/webp-column.webp)
+
+To enable and convert existing media to WebP, run the following commands:
+
+```bash
+wp boilerplate run use-webp-media
+wp boilerplate run regenerate-media
+```
+
+After running these commands, you’ll have your images converted to WebP. Please note that the original files will not be deleted and you’ll have to add additional logic to your Image component for replacing the URLs. Here’s a simple example how to do it:
+
+```php
+// Check if webP is used from admin.
+$isWebPUsed = Components::isWebPMediaUsed();
+
+// If webP is used override the url for large.
+if ($isWebPUsed) {
+	$sourceWebPLarge = Components::getWebPMedia($imageUrl['large'], Media::WEBP_ALLOWED_EXT);
+
+	if ($sourceWebPLarge) {
+		$imageUrl['large'] = $sourceWebPLarge['src'];
+	}
+}
+```
+
+## Dynamic cookies for WP Rocket
+
+This is quite useful when you have a GDPR plugin and you have to take the cookie value into consideration when serving the cached version of the site.
+
+To add the GDPR cookie to the list of dynamic cookies, create a new class called Rocket with the following WP-CLI command:
+
+```bash
+wp boilerplate create service-example --folder=Plugins/Rocket --file_name=Rocket
+```
+
+Add the following method to the Rocket class:
+
+```php
+	/**
+	 * List all dynamic cookies that will create new cached version.
+	 *
+	 * @param array<string> $items Items from the admin.
+	 *
+	 * @return array<string>
+	 */
+	public function dynamicCookiesList(array $items): array
+	{
+		$items[] = 'gdpr';
+
+		return $items;
+	}
+```
+
+Finally, add the following filter to the `register` method:
+
+```php
+\add_filter('rocket_cache_dynamic_cookies', [$this, 'dynamicCookiesList']);
+```
+
+## Yoast SEO helper
+
+The content generated by Eightshift blocks may be difficult for Yoast SEO to interpret, but there is a Yoast SEO helper included in our DevKit that fixes the issue and makes the content readable to Yoast SEO. To enable this, you have to add the following in **_src/Blocks/assets/scripts/application-blocks-editor.js_**:
```

Comment:
Maybe shortly explain here that it's like that because Yoast can't properly read custom dynamic blocks, so we have to tell it how to parse the contents of our blocks.